### ISSUE TITLE
[Java] Implement new Concept Exercise: for-loops

### ISSUE THREAD

**Issue by mirkoperillo** [exercism/v3#2734]
[Use this issue template to describe how to implement a new concept exercises. Please fill in the issue template, and remove any sections wrapped in square brackets (such as this section!).]

This issue describes how to implement the `for-loops` concept exercise for the Java track.

## Getting started

**Please please please read the docs before starting.** Posting PRs without reading these docs will be a lot more frustrating for you during the review cycle, and exhaust Exercism's maintainers' time. So, before diving into the implementation, please read up on the following documents:

- [The features of v3](https://github.com/exercism/v3/blob/master/docs/features-of-v3.md).
- [Rationale for v3](https://github.com/exercism/v3/blob/master/docs/rationale-for-v3.md).
- [What are concept exercise and how they are structured?](https://github.com/exercism/v3/blob/master/docs/concept-exercises.md)

Please also watch the following video:

- [The Anatomy of a Concept Exercise](https://www.youtube.com/watch?v=gkbBqd7hPrA).

## Goal

The goal is to teach a student to work with loops in Java.
This exercise should collect `for loop` and `foreach loop` usage

## Learning objectives

- Know how to use `for loop`
- Know how to use `foreach loop`

## Out of scope

Nothing to report

## Concepts

- `for-loops`

## Prerequisites

To be defined. Prerequisites depend on the exercise implementation

## Resources to refer to

Nothing to report.

### Hints

Nothing to report.

### After

Nothing to report.

## Representer

This exercise does not require any specific representation logic to be added to the [representer](https://github.com/exercism/java-representer).

## Analyzer

This exercise does not require any specific analyzer logic to be added to the [analyzer](https://github.com/exercism/java-analyzer).

## Implementing

To implement this exercise, please [follow these instructions](https://github.com/exercism/v3/blob/master/languages/java/reference/implementing-a-concept-exercise.md).

## Help

If you have any questions while implementing the exercise, please post the questions as comments in this issue.

**Comment by ystromm** [comment]
Hi! I'd like to give this a try!

**Comment by mirkoperillo** [comment]
@ystromm thank you, assigned to you

**Comment by ystromm** [comment]
Any thoughts on the pre-reqs for this exercise? I started on one that uses arrays or lists but I realized that maybe loops should come before arrays in the concepts. Then again foreach-loops require some knowledge of collections.

**Comment by mirkoperillo** [comment]
@ystromm taking C# track as example I found this
```
{
        "slug": "bird-watcher",
        "name": "Bird Watcher",
        "uuid": "b6c532c9-1e89-4fbf-8f08-27f5befb5bb8",
        "concepts": ["arrays", "for-loops", "foreach-loops"],
        "prerequisites": ["booleans", "classes", "if-statements"]
      },
```
I don't see problem having `arrays` as prerequisites of `for-loops`.
For the `foreach-loops` we don't have a `collections` concept currently, what do you think to use `lists`?

**Comment by ystromm** [comment]
Thanks, I'll have a look at the bird-watcher exercise.
